jms/src/main/scala/akka/stream/alpakka/jms/JmsProducerStage.scala

```diff
@@ -7,25 +7,21 @@ package akka.stream.alpakka.jms
 import javax.jms
 import javax.jms.{Connection, Message, MessageProducer, Session}
 
-import akka.Done
-import akka.stream.stage.{GraphStageLogic, GraphStageWithMaterializedValue, InHandler}
 import akka.stream._
+import akka.stream.stage._
 
-import scala.concurrent.{Future, Promise}
-
-final class JmsSinkStage(settings: JmsSinkSettings)
-    extends GraphStageWithMaterializedValue[SinkShape[JmsMessage], Future[Done]] {
+final class JmsProducerStage(settings: JmsProducerSettings) extends GraphStage[FlowShape[JmsMessage, JmsMessage]] {
```

Comment:
Couldn't we use a bounded type for this like you do in the java method, `[T <: JmsMessage]` and have the type flow through nicely so that you get a `Flow[JmsTextMessage, JmsTextMessage, _]` rather than loose the type when you use it?

Reply:
Sure. For my purposes I didn't need the original message any more, so `Unit` would have been good enough.

Reply:
Good, same for the flow factories then. I'd imagine that if you do indeed use it you'd like the type kept.